politeiawww/user/cockroachdb/cockroachdb_test.go

```diff
@@ -0,0 +1,999 @@
+	// Mock data
+	rows := sqlmock.NewRows([]string{"key", "user_id", "created_at", "blob"}).
+		AddRow(sessionKey, session.UserID, session.CreatedAt, eb)
+
+	// Queries
+	sql := `SELECT * FROM "sessions" WHERE "sessions"."key" = $1`
+
+	// Expectations
+	mock.ExpectQuery(regexp.QuoteMeta(sql)).
+		WithArgs(sessionKey).
+		WillReturnRows(rows)
+
+	// Execute method to match queries
+	s, err := cdb.SessionGetByID(session.ID)
+	if err != nil {
+		t.Errorf("session get by id error: %s", err)
+	}
+
+	// Make sure correct session was returned
+	if session.ID != s.ID {
+		t.Errorf("expecting session %s but got %s", session.ID, s.ID)
+	}
+
+	// Make sure query expectations were met
+	err = mock.ExpectationsWereMet()
+	if err != nil {
+		t.Errorf("unfulfilled expectations: %s", err)
+	}
+}
+
+func TestSessionGetByIDFailure(t *testing.T) {
+	cdb, mock, close := setupTestDB(t)
+	defer close()
+
+	// Arguments
+	sessionKey := hex.EncodeToString(util.Digest([]byte("random")))
+
+	// Query
+	sql := `SELECT * FROM "sessions" WHERE "sessions"."key" = $1`
+
+	// Expectations
+	mock.ExpectQuery(regexp.QuoteMeta(sql)).
+		WillReturnError(fmt.Errorf("select sessions error"))
+
+	// Execute method to match queries
+	_, err := cdb.SessionGetByID(sessionKey)
+	if err == nil {
+		t.Errorf("expected error but there was none")
+	}
+
+	// Make sure expectations were met
+	err = mock.ExpectationsWereMet()
+	if err != nil {
+		t.Errorf("unfulfilled expectations: %s", err)
+	}
+}
+
+func TestSessionDeleteByIDSuccess(t *testing.T) {
+	cdb, mock, close := setupTestDB(t)
+	defer close()
+
+	// Arguments
+	session := user.Session{
+		ID:        "1",
+		UserID:    uuid.New(),
+		CreatedAt: time.Now().Unix(),
+		Values:    "",
+	}
+	sessionKey := hex.EncodeToString(util.Digest([]byte(session.ID)))
+	sb, err := user.EncodeSession(session)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+	eb, err := cdb.encrypt(user.VersionSession, sb)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+
+	// Mock data
+	sqlmock.NewRows([]string{"key", "user_id", "created_at", "blob"}).
+		AddRow(sessionKey, session.UserID, session.CreatedAt, eb)
+
+	// Queries
+	sql := `DELETE FROM "sessions" WHERE "sessions"."key" = $1`
+
+	// Expectations
+	mock.ExpectBegin()
+	mock.ExpectExec(regexp.QuoteMeta(sql)).
+		WithArgs(sessionKey).
+		WillReturnResult(sqlmock.NewResult(1, 1))
+	mock.ExpectCommit()
+
+	// Execute method to match queries
+	err = cdb.SessionDeleteByID(session.ID)
+	if err != nil {
+		t.Errorf("session delete by id error: %s", err)
+	}
+
+	// Make sure query expectations were met
+	err = mock.ExpectationsWereMet()
+	if err != nil {
+		t.Errorf("unfulfilled expectations: %s", err)
+	}
+}
+
+func TestSessionDeleteByIDFailure(t *testing.T) {
+	cdb, mock, close := setupTestDB(t)
+	defer close()
+
+	// Query
+	sql := `DELETE FROM "sessions" WHERE "sessions"."key" = $1`
+
+	// Expectations
+	mock.ExpectBegin()
+	mock.ExpectExec(regexp.QuoteMeta(sql)).
+		WillReturnError(fmt.Errorf("delete sessions error"))
```

Comment:
You're hard coding the same error multiple times in some instances.  Create a error variable instead and reference that.

Reply:
Done